smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/readwrite/ReadingClosureUtils.kt

```diff
@@ -22,49 +28,70 @@ class ReadingClosureUtils(
     fun readingClosure(member: MemberShape, isSparse: Boolean = false): String {
         val target = ctx.model.expectShape(member.target)
         val memberTimestampFormatTrait = member.getTrait<TimestampFormatTrait>()
-        val baseClosure = readingClosure(target, memberTimestampFormatTrait)
-        if (isSparse) {
-            return writer.format("optionalFormOf(readingClosure: \$L)", baseClosure)
-        } else {
-            return baseClosure
-        }
+        return makeReadingClosure(target, memberTimestampFormatTrait, isSparse)
     }
 
-    private fun readingClosure(shape: Shape, memberTimestampFormatTrait: TimestampFormatTrait? = null): String {
-        when (shape) {
-            is MapShape -> {
+    private fun makeReadingClosure(shape: Shape, memberTimestampFormatTrait: TimestampFormatTrait? = null, isSparse: Boolean): String {
+        val base = when {
+            shape is MapShape -> {
```

Comment:
Changed this logic a bit so that I can write the new reading closures correctly.